server/lib/chat/chat-service.ts

```diff
@@ -92,6 +93,10 @@ export default class ChatService {
       throw new ChatServiceError(ChatServiceErrorCode.InvalidJoinAction, 'Already in this channel')
     }
 
+    const userInfo = await findUserById(userId)
```

Comment:
I think as a slight optimization you could await this at the same time as `isUserBannedFromChannel` below, e.g.
`const [userInfo, isBanned] = await Promise.all([
  findUserById(userId),
  isUserBannedFromChannel(originalChannelName, userId),
])`

not really a huge deal but reduces a round trip to the DB